### ISSUE TITLE
bug: radio button disabled prop doesn't handle undefined

### ISSUE THREAD

**Issue by nmspackman** [ionic-team/ionic-framework#28677]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

An `IonRadio` component with `undefined` passed as a value to the `disabled` prop causes the radio to not trigger the onIonChange function when selected. The visual effect is the previous selection and the current selection (with disabled set to undefined) will **both** be selected.'

Note: Cherries is the radio that has `disabled` set to `undefined`.
<img width="232" alt="image" src="https://github.com/ionic-team/ionic-framework/assets/11526734/d526be20-0485-40e6-bddc-aada56b9463b">

### Expected Behavior

The radio button should be selected and the radio group value should update.

### Steps to Reproduce

1. Create an `IonRadioGroup` with 3 `IonRadio` components nested (values are arbitrary)
2. Set one of the `IonRadio`'s disabled prop to undefined
3. When viewing the UI, click one of the radios without the disabled prop set, then click the radio with it set

### Code Reproduction URL

https://stackblitz.com/edit/f4avhg?file=src%2Fmain.tsx

### Ionic Info

N/A

### Additional Information

To work around this issue, I set the `disabled` prop using strict equals for comparing the variable passed in. It looks something like this:
```
<IonRadio value="cherries" disabled={options?.isDisabled === true}>
    Cherries
</IonRadio>
```

**Comment by liamdebeasi** [comment]
Thanks for the report. Can you help me understand why you need to pass `undefined` instead of `false` to `disabled`? As per the docs, the `disabled` property accepts a boolean: https://ionicframework.com/docs/api/radio#properties This behavior is consistent with the other `disabled` properties in Ionic.

**Comment by nmspackman** [comment]
Thanks for the quick response!

I wouldn't say _need_, but that's how we previously were using the component without issue. Typically the value would be true or false, but it was possible for that value to not be set (undefined). We upgraded, then caught this behavior weeks later. So, it appeared to be a bug.

The only confusing thing as a TS developer is that (despite the type being boolean) being optional allows for undefined to be passed in. Passing in any other falsy value of course will be caught, but undefined is the only exception. My personal expectation is that any prop set to undefined would function as if the prop was never passed at all.

At the end of the day, it's not a big deal. We now know, but others may benefit from it.

**Comment by liamdebeasi** [comment]
> The only confusing thing as a TS developer is that (despite the type being boolean) being optional

Out of curiosity, where are you seeing that this property is marked as optional? The intent of the `disabled` property is that it is required, but maybe there's a type mixup somewhere.

**Comment by nmspackman** [comment]
`@ionic/core/dist/types/components.d.ts`:
<img width="409" alt="image" src="https://github.com/ionic-team/ionic-framework/assets/11526734/6f65f4d2-8a5c-4b7c-8fd6-e3951510720d">

Which shows in the VSCode tooltip as:
<img width="468" alt="image" src="https://github.com/ionic-team/ionic-framework/assets/11526734/4dc12b3f-78cd-42dc-9bf2-2ab13b7880d1">

**Comment by HudsonPWesel** [comment]
Okay so I was able to fix it by putting cherries first still working on why this behavior occurs
        <IonRadio value="cherries" >Cherries</IonRadio>
        <br />
        <IonRadio value="strawberries">Strawberries</IonRadio>
        <br />
        <IonRadio value="pineapple">Pineapple</IonRadio>
        <br />
        <IonRadio value="grapes">Grapes</IonRadio>
      </IonRadioGroup>

**Comment by HudsonPWesel** [comment]
You need a <br /> before each Ion element

**Comment by HudsonPWesel** [comment]
You could create a wrapper IonGroup and force the type like this:
` interface IonProps {
    value: string;
    disabled: boolean;
  }
  const CustomIonRadioGroup: React.FC<IonProps> = (props: IonProps) => {
    return (
      <>
        <p>Current Value: {value}</p>
        <IonRadioGroup
          value={value}
          onIonChange={(e) => {
            console.log(value);
            setValue(e.detail.value);
          }}
        >
          <IonRadio value="cherries" disabled={props.disabled}>
            Cherries
          </IonRadio>
          <br />
          <IonRadio value="grapes">Grapes</IonRadio>
          <br />
          <IonRadio value="strawberries">Strawberries</IonRadio>
          <br />
          <IonRadio value="pineapple">Pineapple</IonRadio>
          <br />
        </IonRadioGroup>
      </>
    );
  };
  return <CustomIonRadioGroup value={value} disabled={false} />;`

**Comment by 1pandeyji** [comment]
I can solve this issue.

**Comment by liamdebeasi** [comment]
> `@ionic/core/dist/types/components.d.ts`: <img alt="image" width="409" src="https://private-user-images.githubusercontent.com/11526734/289223794-6f65f4d2-8a5c-4b7c-8fd6-e3951510720d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzMDQyMTUsIm5iZiI6MTcwMjMwMzkxNSwicGF0aCI6Ii8xMTUyNjczNC8yODkyMjM3OTQtNmY2NWY0ZDItOGE1Yy00YjdjLThmZDYtZTM5NTE1MTA3MjBkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDE0MTE1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3YzkzZWIzNzJkYmUyNjc5M2E4MTUwMGM4YWFhMTdmN2U1M2UzMmI3NTNhNjMwOTM4NmRkN2FlZTNhOGY4ZmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EGZ4i--KFAs3Led4oz7Eh5G-5biuIl6yJG0vmqw2NMU">
> 
> Which shows in the VSCode tooltip as: <img alt="image" width="468" src="https://private-user-images.githubusercontent.com/11526734/289223743-4dc12b3f-78cd-42dc-9bf2-2ab13b7880d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIzMDQyMTUsIm5iZiI6MTcwMjMwMzkxNSwicGF0aCI6Ii8xMTUyNjczNC8yODkyMjM3NDMtNGRjMTJiM2YtNzhjZC00MmRjLTliZjItMmFiMTNiNzg4MGQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjExVDE0MTE1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2NWRmMjIxMGRhYmFkNWU4MDQ4YjJiYjU5NTg5MDA2NzE2MjcyNGY4N2ZlMDcwZmRiZmE5NWFmNTk4N2JlOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PnQbypRYDNXGvUc66lSnFeGhSldT1ll-Y-oKp2VBIR4">

Thanks for the info! I need to check with the team to see if having the `disabled` prop be optional is intentional. The team's course of action may change based on that discussion. I'll post here when I have more to share.

**Comment by liamdebeasi** [comment]
Hey there,

I checked with the team, and having `disabled` be optional was a mistake on our end. I'll look into fixing this, but the change will need to be in a major release of Ionic since it is a breaking change.

**Comment by liamdebeasi** [comment]
Hi everyone,

I spoke with the team and having `disabled` be optional does not appear to be intentional. However, most of our components that use `disabled` are built in such a way that they support the optional use case. Additionally, making this property required would mean we'd need to change the types for 28 components, meaning 28 breaking changes. We don't think that is a good tradeoff to make, so we'll focus on updating radio to resolve this issue.

**Comment by mapsandapps** [comment]
Thanks for submitting this issue! This has been fixed in #28712 and will be released in an upcoming version of Ionic Framework.
